### PR TITLE
[OSF-7480] Skip Guids with missing/invalid referents

### PR DESCRIPTION
**Pull request by sloria**
scripts.mark_orphaned_guids marked invalid Guids with is_orphaned=True.
Now we can skip over those Guids when we migrate the Guid collection

## Ticket
https://openscience.atlassian.net/browse/OSF-7480

### CODE REVIEW

File: osf/management/commands/migratedata.py [hunk 1]
```diff
@@ -333,8 +332,10 @@ def fix_guids():
     short_missing = [x for x in missing if len(x) < 6]
     long_missing = [x for x in missing if len(x) > 5]
     assert len(short_missing) + len(long_missing) == len(missing), 'It broke'
-    short_missing_guids = MGuid.find(MQ('_id', 'in', short_missing))
-    long_missing_guids = MGuid.find(MQ('_id', 'in', long_missing))
+    # NOTE: is_orphaned will be True for Guids that should not be migrated because they have
+    # invalid/missing referents
+    short_missing_guids = MGuid.find(MQ('_id', 'in', short_missing) & MQ('is_orphaned', 'ne', True))
```
Comment (cwisecarver) [inline osf/management/commands/migratedata.py]: This should not find anything, correct? In theory after your script runs the ophaned guids will be marked at skipped here, correct?

**Review by cwisecarver — APPROVED** [review]
Just the one question. lgtm

File: osf/management/commands/migratedata.py (hunk 1 above)
Comment (sloria) [reply osf/management/commands/migratedata.py]: Correct. So next time you migrate the prod db, take note of any `"Guid 123ab does not match..."` messages; they are not expected.

**Comment by sloria** [comment]
I've run the script to mark the invalid GUIDs on prod, so this change should take effect next time you migrate a copy of prod.

**Comment by sloria** [comment]
Closing in favor of https://github.com/cwisecarver/osf.io/pull/9

**Pull request closed without merging**